Published-source: https://deb.freexian.com/extended-lts/pool/main/p/php7.0/php7.0_7.0.33-0+deb9u26.debian.tar.xz
Published-patch: 0059-Fix-bug-77396-Null-Pointer-Dereference-in-phar_creat.patch
Scope: selected PHP runtime components and their tests; packaging and unrelated extensions omitted.

From: Stanislav Malyshev <stas@php.net>
Date: Sun, 3 Mar 2019 18:22:32 -0800
Subject: Fix bug #77396 - Null Pointer Dereference in
 phar_create_or_parse_filename

(cherry picked from commit 7f0ab7c20c83a52862ad7c8acf31c3fa739f1274)
---
 ext/phar/phar.c              |  3 +++
 ext/phar/tests/bug77396.phpt | 15 +++++++++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 ext/phar/tests/bug77396.phpt

--- a/ext/phar/phar.c
+++ b/ext/phar/phar.c
@@ -1390,6 +1390,9 @@ int phar_create_or_parse_filename(char *fname, int fname_len, char *alias, int a
 	/* set up our manifest */
 	mydata = ecalloc(1, sizeof(phar_archive_data));
 	mydata->fname = expand_filepath(fname, NULL);
+	if (mydata->fname == NULL) {
+		return FAILURE;
+	}
 	fname_len = strlen(mydata->fname);
 #ifdef PHP_WIN32
 	phar_unixify_path_separators(mydata->fname, fname_len);
--- /dev/null
+++ b/ext/phar/tests/bug77396.phpt
@@ -0,0 +1,15 @@
+--TEST--
+Bug #77396 Relative filename exceeding maximum path length causes null pointer dereference.
+--SKIPIF--
+<?php if (!extension_loaded("phar")) die("skip"); ?>
+--FILE--
+<?php
+$path = '../' . str_repeat("x", PHP_MAXPATHLEN) . '.tar';
+$phar = new PharData($path);
+?>
+--EXPECTF--
+Fatal error: Uncaught UnexpectedValueException: Phar creation or opening failed in %s/bug77396.php:%d
+Stack trace:
+#0 %s/bug77396.php(%d): PharData->__construct(%s)
+#1 {main}
+  thrown in %s/bug77396.php on line %d
